pkg/scheduler/scheduler_test.go

```diff
@@ -1153,6 +1209,12 @@ func TestSchedule(t *testing.T) {
 			if diff := cmp.Diff(tc.wantInadmissibleLeft, qDumpInadmissible); diff != "" {
 				t.Errorf("Unexpected elements left in inadmissible workloads (-want,+got):\n%s", diff)
 			}
+
+			if len(tc.wantEvents) > 0 {
```

Comment:
Can we drop this check, and always compare? It may detect us some wrong events where we didn't expect them (so we left `wantEvents empty).

Reply:
There are 19 tests and 48 events to take into account , some of them might not be in a strict order ,  and not directly related to Admission/QuotaReservation.

Let's open a follow-up issue for this if necessary.

Reply:
sgtm